docs/source/contributor-guide/index.md

```diff
@@ -216,3 +216,23 @@ The good thing about open code and open development is that any issues in one ch
 
 Pull requests will be marked with a `stale` label after 60 days of inactivity and then closed 7 days after that.
 Commenting on the PR will remove the `stale` label.
+
+## Building
```

Comment:
Thank you @dentiny  -- I think most of this information is already present in this page:

https://datafusion.apache.org/contributor-guide/getting_started.html

Would you be willing to combine your contribution with the existing information there?

Reply:
> I think most of this information is already present in this page:

Ohhhh my bad! I didn't find it's the subpage for contributing page...

Reply:
I removed instructions for build, they're already included in the existing dev env doc;
Keep some sentence for setting dev container for easier operation.

Reply:
In the latest commit, I remove those unrelated.